### ISSUE TITLE
Task list views not filtering - need example

### ISSUE THREAD

**Issue by BECKYLEA11** [tomansley/SimpliListViews#37]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

**Comment by tomansley** [comment]
Hello,

I am going to need more details before I can move forward with looking at the issue. Can you provide screenshots of standard list view page and the SLV page? Are some filters working and others are not?

I am looking at the Task SLV list views and they look to be filtering records appropriately based on the standard list views.

Thanks
Tom

**Comment by BECKYLEA11** [comment]
I think the only issue I have is what join field value to use as Tasks use the related to or WhatId  and I have tried every permutation to try and sort on this. My example is a Custom Object called Project and I am using Tasks. I would like the tasks to filter when using a two list view or on a record page within a tab. The problem is I get all Tasks and not just the project tasks for the record in both instances. I will attach a screen shot without the join field which seems to be my only issue I think.

**Comment by BECKYLEA11** [comment]
![app proj 3](https://user-images.githubusercontent.com/93476225/139592218-e1a16cf9-ffc1-4698-8179-447e7d9a4ef3.png)
![app proj 2](https://user-images.githubusercontent.com/93476225/139592221-de72fa4c-ccf8-47e4-bf86-0e9288f1b2a6.png)
![pro app 1](https://user-images.githubusercontent.com/93476225/139592223-a4ce656f-8e48-4ad9-acf4-6fae487f2e34.png)
![Proj rec page 2](https://user-images.githubusercontent.com/93476225/139592225-e0d23970-56cf-44b9-9547-ffc9a8ae4e32.png)
![project rec page 1](https://user-images.githubusercontent.com/93476225/139592226-998007af-0ea7-4b29-acbc-cdb81af9515d.png)

**Comment by BECKYLEA11** [comment]
Thank you for looking at this - I love the app and am trying to get it
working for our project implementation live Monday. If you can
get me up and going I will give an awesome review!

**Comment by BECKYLEA11** [comment]
And also the status field will not inline edit - does have anything to do with being a required field?

**Comment by tomansley** [comment]
The "Join Criteria Data" field is only for using the component in code and
you want to pass data into the component. You don't want to use this.

You want to use the "Joined Field Name". You also want to make sure you set
"Use Message Channel" to checked because that is what identifies that the
component should be listening for checked records from the other component.

Does that help you resolve your issues?

**Comment by BECKYLEA11** [comment]
That is helpful but no it does not unfortunately. The joined field name seems to be the issue. My custom object api name is Project__c - I haven't been able to get any of the tasks or other custom related object records to filter on the selected project record on the app or on the record page component. I am not sure if it is the formatting which if you could give me an example it might be helpful. The videos don't show that detail. The task is a tricky one as the related to field is the link and the whatid is the api name. I have a related object called issues/risks that is joined by the project field project__c which again does not seem to filter the records. Perhaps you can send me an screen shot of a custom object and related records as well as a custom object with task joined field names. Is that possible?
I have spent so many hours trying to do this I was hoping it would be much simpler than it seems to be. Any assistance would be appreciated.

**Comment by tomansley** [comment]
Ok, I figured out the issue. You are trying to join the two sets of data on a field (RelatedId) which is polymorphic. i.e. it can take an Id which is of many different types. The code is not written to handle this. It is an unusual case as most lookup fields are for a specific object type. But, there is no reason it shouldn't be able to.

I played around with the code and got it working as expected which you can see from the screenshot below.

I can get an updated BETA package out tomorrow to resolve this issue. I am on CST time so it should go out at about 9AM CST. Not sure if that is soon enough for you to get things working before your demo. The production ready code would go out in the next release. This means you would only be able to run the package in a sandbox though.

![Screen Shot 2021-10-31 at 7 34 17 PM](https://user-images.githubusercontent.com/375321/139607219-32977b39-0deb-493d-ad7f-2bbf35df481a.png)

**Comment by BECKYLEA11** [comment]
Great news that you were able to get an answer and update the product. I think that is a good enhancement. We are going live with projects tomorrow but I think we can work around it with using the search filter in the list view for a bit. When do you expect the new release to be out? Any other ideas for work arounds in the meantime? Can you screen shot for me the record page component properties so that I can add to the sandbox correctly? Thanks so much for getting back to me. It saved me time knowing its not within the capabilities at this point in time.

**Comment by BECKYLEA11** [comment]
Oh also I didn't see the capability to export the list view even though the box was ticked as part of the component properties.

**Comment by tomansley** [comment]
I am hoping for the end of this week. 


On Oct 31, 2021 20:22, BECKYLEA11 ***@***.***> wrote: 

Great news that you were able to get an answer and update the product. I think that is a good enhancement. We are going live with projects tomorrow but I think we can work around it with using the search filter in the list view for a bit. When do you expect the new release to be out? Any other ideas for work arounds in the meantime? Can you screen shot for me the record page component properties so that I can add to the sandbox correctly? Thanks so much for getting back to me. It saved me time knowing its not within the capabilities at this point in time.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by BECKYLEA11** [comment]
Is it part of your app that picklists can't be edited inline?

**Comment by tomansley** [comment]
Most picklists can be edited inline. 


For some though Salesforce doesn't provide all the data we need which is unfortunate. I believe the ones they don't provide are all standard fields and I do believe I saw some on the Task object. 


I could look at any picklists values you are struggling with to determine if I could get them another way, but that won't get released for a while. 


On Oct 31, 2021 21:17, BECKYLEA11 ***@***.***> wrote: 

Is it part of your app that picklists can't be edited inline?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by tomansley** [comment]
Ok, here is the test app link to use to download the Task field changes in
a SANDBOX -
https://<SANDBOX_URL>.salesforce.com/packaging/installPackage.apexp?p0=04t3h0000043tS9

Do not push to PROD. It is not ready yet.

I have also attached the config I used to create the scenario.

*What the page looks like when running*
[image: image.png]

*Next 2 are the PROJECT LIST VIEW CONFIG*
[image: image.png]

[image: image.png]

-------------------------------------------------------------------------------------------------------------------------------------


*NEXT 2 ARE TASK LIST VIEW CONFIG*
[image: image.png]

[image: image.png]
